test/unit/core_test_utils.c

```diff
@@ -32,6 +32,7 @@
 /**
  * Mock function to resolve circular dependencies in unit tests
  */
-void cord_on_yield(void)
+void __attribute__((weak))
+cord_on_yield(void)
```

Comment:
This doesn't look right. I'd expect the core version of `cord_on_yield` to be defined as weak. OTOH I'm not sure about possible implications of such a change so I'd avoid weak linkage altogether if possible.

Anyway, it's unclear what exactly you're trying to fix here. Is there a build failure on some platform?